examples/chat/module_0/rust/src/lib.rs

```diff
@@ -14,159 +14,133 @@
 // limitations under the License.
 //
 
-use chat_common::proto::chat::{CreateRoomRequest, DestroyRoomRequest};
-use chat_common::proto::chat::{JoinRoomRequest, ReceivedMessage, SentMessage};
+use chat_common::command::Command;
+use chat_common::proto::chat::{
+    CreateRoomRequest, DestroyRoomRequest, SendMessageRequest, SubscribeRequest,
+};
 use chat_common::proto::chat_grpc::{dispatch, ChatNode};
 use log::info;
 use oak::grpc;
 use oak::grpc::OakNode;
 use oak_derive::OakExports;
 use protobuf::well_known_types::Empty;
 use protobuf::{Message, ProtobufEnum};
+use std::collections::hash_map::Entry;
 use std::collections::HashMap;
 
 type RoomId = Vec<u8>;
-type AdminId = Vec<u8>;
+type AdminToken = Vec<u8>;
 
-#[derive(OakExports)]
+#[derive(OakExports, Default)]
 struct Node {
-    // TODO: use bimap crate for these fields?
-    room_to_admin: HashMap<RoomId, AdminId>,
-    admin_to_room: HashMap<AdminId, RoomId>,
-    rooms: HashMap<AdminId, Room>,
+    rooms: HashMap<RoomId, Room>,
 }
 
 struct Room {
-    name: String,
     channel: oak::WriteHandle,
+    admin_token: AdminToken,
+}
+
+impl Room {
+    fn new(admin_token: AdminToken) -> Self {
+        let (wh, rh) = oak::channel_create().unwrap();
+        oak::node_create("room-config", rh);
+        oak::channel_close(rh.handle);
+        Room {
+            channel: wh,
+            admin_token,
+        }
+    }
 }
 
 impl OakNode for Node {
     fn new() -> Self {
         oak_log::init_default();
-        Node {
-            room_to_admin: HashMap::new(),
-            admin_to_room: HashMap::new(),
-            rooms: HashMap::new(),
-        }
+        Node::default()
     }
     fn invoke(&mut self, method: &str, req: &[u8], writer: grpc::ChannelResponseWriter) {
```

Comment:
If the standard method for sdk usage is to have invoke calling dispatch, perhaps the `#[derive(OakExports)]` should also generate an invoke method, completely hiding this from the user. At that point we are in quite a good place in terms of hiding complexity to the developer. If we require `#[derive(OakExports,Default)]` for the SDK as per this example we could derive `new` as well. This isn't too crazy as we expect Nodes to configure themselves after reading values from their initial channel.

Reply:
Yes we should consider that. Feel free to try out new APIs / patterns if you have any ideas, and we can discuss in a PR or issue.